feat: add scheduled markdown link check workflow

drtoxic69 opened the pull request:
**Description**:
This PR introduces a new GitHub Actions workflow to automatically check for broken links in Markdown files on a monthly schedule. It ensures documentation health by reporting failures via an automated GitHub Issue.

* Add `.github/workflows/cron-check-broken-links.yml` to run checks monthly or manually.
* Implement idempotent issue reporting using `actions/github-script` (creates a new issue or updates an existing one to avoid duplicates).
* Update `CHANGELOG.md` with the new feature.

**Related issue(s)**:

Fixes #1210 

**Notes for reviewer**:
I have implemented the reporting logic using `actions/github-script` to avoid introducing new third-party dependencies.

**Important Setup Note:**
The workflow uses the labels `broken-markdown-links` and `automated` to manage issues. Please ensure these labels exist in the repository.

**Key Features:**
1. **Security:** All Actions use pinned SHAs matching the existing `pr-check-broken-links.yml` workflow.
2. **Idempotency:** The reporting script checks for an existing open issue with the label `broken-markdown-links`.
    * If found: It adds a comment ("Still broken").
    * If not found: It creates a new issue.
3. **Manual Testing:** The workflow includes a `workflow_dispatch` trigger with a `dry_run` input to test the logic without spamming issues.

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

exploreriii reviewed (changes requested):
Hi @drtoxic69 looking great!
Would you mind testing this on your fork? You'll need to change the cron job so you don't wait a month 👍 
docs/sdk_developers/training/testing_forks.md

just merge it to your main, enable issue creaiton in your settings page, and see if it creates it or logs anything in the actions tab
see similar example:
https://github.com/exploreriii/hiero_sdk_python/issues

drtoxic69 commented:
I have verified this workflow on my fork, and it is working correctly.

## **Testing Steps Performed:**
1. Enabled Issues on my fork.
2. Intentionally added a broken link to `README.md`.
3. Manually triggered the workflow with `dry_run: false`.
4. **Result:** The workflow correctly detected the failure and created the Issue with the appropriate labels.

**Implementation Note:**
During testing, I switched the link checker from `tcort/markdown-link-check` (Docker-based) to `lycheeverse/lychee-action` (Native). The Docker-based action was having trouble mounting the volume and discovering files in this runner environment. Lychee is faster, runs natively, and reliably detects the files.

## **Screenshots:**
_Action Run Log (Detecting the broken link):_
<img width="1661" height="1053" alt="Screenshot 2025-12-27 at 10 09 04 PM" src="https://github.com/user-attachments/assets/6c95890f-bac1-46d4-b429-fdae52b9c663" />
_Automated Issue Created:_
<img width="1296" height="538" alt="Screenshot 2025-12-27 at 10 09 59 PM" src="https://github.com/user-attachments/assets/4dfcb8cb-2017-486f-9a0f-0bb1aa13b975" />

I have squashed the history into a single clean commit and reverted the `cron` schedule to run monthly. Ready for review!

exploreriii commented on .github/workflows/cron-check-broken-links.yml:
```diff
@@ -0,0 +1,130 @@
+
+      - name: Check Markdown links (Lychee)
+        id: lychee
+        uses: lycheeverse/lychee-action@22134d37a1fff6c2974df9c92a7c7e1e86a08f9c # v1.9.0
```
Am I correct thinking  the latest version is 2.7?
https://github.com/lycheeverse/lychee-action/releases/tag/v2.7.0

exploreriii commented on .github/workflows/cron-check-broken-links.yml:
what is core and are you importing it?

drtoxic69 replied:
Since I'm using the `actions/github-script` action, it automatically provides `core` as a global variable. We don't need to manually import it.

You can see the list of automatically included variables in the [official documentation](https://github.com/actions/github-script?tab=readme-ov-file#available-globals).

drtoxic69 commented on .github/workflows/cron-check-broken-links.yml:
okay, I will be removing this part.
